receiver/k8sclusterreceiver/collection/collector.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collection
+
+import (
+	"reflect"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/autoscaling/v2beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+)
+
+// TODO: Consider moving some of these constants to
+// https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/conventions/opentelemetry.go.
+
+// Resource label keys.
+const (
+	// Resource Type
+	k8sType       = "k8s"
+	containerType = "container"
+
+	// Resource labels keys for UID.
+	k8sKeyPodUID                   = "k8s.pod.uid"
+	k8sKeyNodeUID                  = "k8s.node.uid"
+	k8sKeyCronJobUID               = "k8s.cronjob.uid"
+	k8sKeyDeploymentUID            = "k8s.deployment.uid"
+	k8sKeyJobUID                   = "k8s.job.uid"
+	k8sKeyNamespaceUID             = "k8s.namespace.uid"
+	k8sKeyReplicaSetUID            = "k8s.replicaset.uid"
+	k8sKeyReplicationControllerUID = "k8s.replicationcontroller.uid"
+	k8sKeyStatefulSetUID           = "k8s.statefulset.uid"
+	k8sKeyDaemonSetUID             = "k8s.daemonset.uid"
+	k8sKeyHPAUID                   = "k8s.hpa.uid"
+	k8sKeyResourceQuotaUID         = "k8s.resourcequota.uid"
+
+	// Resource labels keys for Name.
+	k8sKeyCronJobName               = "k8s.cronjob.name"
+	k8sKeyNodeName                  = "k8s.node.name"
+	k8sKeyDeploymentName            = "k8s.deployment.name"
+	k8sKeyJobName                   = "k8s.job.name"
+	k8sKeyReplicaSetName            = "k8s.replicaset.name"
+	k8sKeyReplicationControllerName = "k8s.replicationcontroller.name"
+	k8sKeyStatefulSetName           = "k8s.statefulset.name"
+	k8sKeyDaemonSetName             = "k8s.daemonset.name"
+	k8sKeyHPAName                   = "k8s.hpa.name"
+	k8sKeyResourceQuotaName         = "k8s.resourcequota.name"
+
+	// Resource labels for container.
+	containerKeyID       = "container.id"
+	containerKeySpecName = "container.spec.name"
```

Comment:
This should be just `"container.name"`. See https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#container

Reply:
This is the name of the container provided in a Pod spec. I don't think this will reflect the name of the the container that's actually running. If I understand correctly `container.name` should be the name of the actual running container, right?

Reply:
Not sure I understand. Can you provide an example of the config + the name that will be create, something from real kubernetes.